proposals/lip-define_state_and_state_transition_of_token_module.md

```diff
@@ -0,0 +1,1287 @@
+This is for example used in all input formats for the module's exposed functions.
+This choice follows a potential way the module could be implemented in JavaScript, the same behavior could be implemented with a named tuple in Python.
+This allows the exposed function interfaces to be simple and uniform.
+
+
+#### Token ID and Native Tokens
+
+Tokens on their native chain are identified by the pair `{"chainID": CHAIN_ID_ALIAS_NATIVE, "localID": localID}`. 
+The same tokens in other chains would be identified by the pair `{"chainID": nativeChainID, "localID": localID}`. 
+
+In all sidechains, the LSK token is identified by the pair `{"chainID": 1, "localID": 0}`, i.e., `chainID = MAINCHAIN_ID = 1` and `localID = 0`. 
+This is in contrast with the LSK ID on mainchain which is `{"chainID": 0, "localID": 0}`.
+
+
+#### Supported Tokens
+
+The Token module contains a function used when receiving cross-chain messages to assert the support for non-native tokens. 
+It should return a boolean, depending on the configuration of the Token module.
+For the rest of this LIP, this function is written `tokenSupported(tokenID)`.
+It must satisfy the condition below:
+
+*   `tokenSupported({"chainID": MAINCHAIN_ID, "localID":  LOCAL_ID_LSK}) = True`. This corresponds to the token ID of the LSK token.
+
+Further, on the Lisk mainchain, the LSK token is the only supported token (no tokens with different chain ID are supported).
+
+
+### Token Module Store
+
+The Token module store is separated in four parts, the supply substore, the escrow substore, the terminated escrow substore and the user substore. 
+
+
+#### Supply Substore
+
+The Token module store contains an entry dedicated to storing information about the total supply of native tokens. 
+The substore contains entries with:
+
+
+*   The store prefix is set to `STORE_PREFIX_SUPPLY`.
+*   Each store key is a serialized local ID: `uint32be(localID)`.
+*   Each store value is the serialization of an object following `supplyStoreSchema`.
+
+```java
+supplyStoreSchema = {
+    "type": "object",
+    "required": ["totalSupply"],
+    "properties": {
+        "totalSupply": { 
+            "dataType": "uint64",
+            "fieldNumber": 1
+        },
+    }
+}
+```
+The default value for this substore is `{"totalSupply": 0}` serialized using `supplyStoreSchema`. 
+
+
+#### Available Local ID Substore
+
+The Token module store contains an entry dedicated to storing information about the available local IDs:
+
+
+*   The store prefix is set to `STORE_PREFIX_AVAILABLE_LOCAL_ID`.
+*   Each store key is the empty bytes.
+*   Each store value is the serialization of an object following `availableLocalIDStoreSchema`.
+
+```java
+availableLocalIDStoreSchema = {
+    "type": "object",
+    "properties": {
+        "nextAvailableLocalID": { 
+            "dataType": "uint32",
+            "fieldNumber": 1
+        },
+    },
+    "required": ["nextAvailableLocalID"]
```

Comment:
This schema and the following two were not changed. Note also that some closing commas need to be removed here and for the following two.

Reply:
thanks done